client/src/components/dialogs/AvatarPicker.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import './avatar-picker.css';
+
+class AvatarPicker extends React.Component {
```

Comment:
@existentialcoder Please try to use Functional components as much as possible as it goes with the current style of code + React also recommends the same.

Reply:
In functional components I faced an infinite loop problem! - https://stackoverflow.com/questions/59112809/react-usestate-hook-causing-infinite-loop

Reply:
@mahadevans87 : Changed it to functional components. 
@existentialcoder : Nice catch. Like it is mentioned in the above stackoverflow link, we have to check the avatars array before hitting the URL. If it is present, we need not fetch it again, hence preventing a re-render.